server/service/typedb_service.rs

```diff
@@ -156,6 +162,67 @@ impl typedb_protocol::type_db_server::TypeDb for TypeDBService {
             .map_err(|err| err.into_error_message().into_status())
     }
 
+    async fn users_get(
+        &self,
+        request: Request<typedb_protocol::user_manager::get::Req>,
+    ) -> Result<Response<typedb_protocol::user_manager::get::Res>, Status> {
+        let get_req = request.into_inner();
+        let user = self.user_manager.get(get_req.name.as_str());
+        match user {
+            Some((u, _)) => Ok(Response::new(users_get_res(u))),
+            None => Err(ServiceError::UserDoesNotExist { name: get_req.name }.into_error_message().into_status()),
+        }
+    }
+
+    async fn users_all(
+        &self,
+        _: Request<typedb_protocol::user_manager::all::Req>,
+    ) -> Result<Response<typedb_protocol::user_manager::all::Res>, Status> {
+        let users = self.user_manager.all();
+        Ok(Response::new(users_all_res(users)))
+    }
+
+    async fn users_contains(
+        &self,
+        request: Request<typedb_protocol::user_manager::contains::Req>,
+    ) -> Result<Response<typedb_protocol::user_manager::contains::Res>, Status> {
+        let contains_req = request.into_inner();
+        let contains = self.user_manager.contains(contains_req.name.as_str());
+        Ok(Response::new(users_contains_res(contains)))
+    }
+
+    async fn users_create(
+        &self,
+        request: Request<typedb_protocol::user_manager::create::Req>,
+    ) -> Result<Response<typedb_protocol::user_manager::create::Res>, Status> {
+        users_create_req(request)
+            .and_then(|(usr, cred)| self.user_manager.create(&usr, &cred))
+            .map(|_| Response::new(user_create_res()))
+            .map_err(|err| err.into_error_message().into_status())
+    }
+
+    async fn users_update(
+        &self,
+        request: Request<typedb_protocol::user::update::Req>,
+    ) -> Result<Response<typedb_protocol::user::update::Res>, Status> {
+        users_update_req(request)
+            .and_then(|(username, user, cred)| self.user_manager.update(username.as_str(), &user, &cred))
+            .map(|_| Response::new(user_update_res()))
+            .map_err(|err| err.into_error_message().into_status())
+    }
+
+    async fn users_delete(
```

Comment:
Where is/will be the validation of your rights to delete a user, of the users available for deletion?

Reply:
> Can we delete any user?
Good catch. I've added protection against deleting the default user here: a710dc0564f59ba97e860d96d25b42cb2f1df12e. In the future we'll aim for a better solution since this solution isn't very flexible / user friendly.

> Where is/will be the validation of your rights to delete a user, of the users available for deletion?
This is when we implement roles